WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/FollowersUseCase.kt

```diff
@@ -119,35 +128,49 @@ class FollowersUseCase
                                     R.string.stats_followers_wordpress_com,
                                     R.string.stats_followers_email
                             ),
-                            uiState
-                    ) {
-                        onUiState(it)
+                            uiState.selectedTab
+                    ) { selectedTab ->
+                        updateUiState { it.copy(selectedTab = selectedTab) }
                     }
             )
-            if (uiState == 0) {
+            if (uiState.selectedTab == 0) {
                 items.addAll(buildTab(wpComModel, R.string.stats_followers_wordpress_com))
             } else {
                 items.addAll(buildTab(emailModel, R.string.stats_followers_email))
             }
 
-            if (wpComModel.hasMore && uiState == 0 || emailModel.hasMore && uiState == 1) {
-                val buttonText = if (useCaseMode == VIEW_ALL)
-                        R.string.stats_insights_load_more
-                    else
-                        R.string.stats_insights_view_more
-                items.add(
-                        Link(
-                                text = buttonText,
-                                navigateAction = NavigationAction.create(uiState, this::onLinkClick)
-                        )
-                )
+            if (wpComModel.hasMore && uiState.selectedTab == 0 || emailModel.hasMore && uiState.selectedTab == 1) {
+                if (useCaseMode == BLOCK) {
+                    val buttonText = R.string.stats_insights_view_more
+                    items.add(
+                            Link(
+                                    text = buttonText,
+                                    navigateAction = NavigationAction.create(uiState.selectedTab, this::onLinkClick)
+                            )
+                    )
+                } else {
+                    items.add(LoadingItem(this::loadMore, isLoading = uiState.isLoading))
+                }
             }
         } else {
             items.add(Empty())
         }
         return items
     }
 
+    private fun loadMore() {
+        GlobalScope.launch(bgDispatcher) {
+            withContext(mainDispatcher) {
+                updateUiState { it.copy(isLoading = true) }
```

Comment:
Is this still necessary now that the UI state is updated inside `fetchData()`?

Reply:
ah you're right, I didn't realise `loadMore` calls `fetchData`, good call 👍